### PR TITLE
Fixes #2963: upgrade Quarkus from 3.6.1 to the latest (3.12.0) version to resolve CVEs

### LINKED ISSUE #2963: Update Quarkus from 3.6 to 3.12 to resolve CVEs
Looks like we are using Quarkus 3.6(.3), released about 6 months ago. It has more than dozen CVEs (for dependencies). It'd make sense to upgrade to a later version (like 3.12.0) which has some or all of them resolved:

https://mvnrepository.com/artifact/io.quarkus/quarkus-bom

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Upgrades Quarkus used by APIs to a later version, to resolve CVEs reported against dependencies and allow use of later Jackson version by Data API.

**Which issue(s) this PR fixes**:
Fixes #2963

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/CQLResource.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
 @SecurityRequirement(name = RestOpenApiConstants.SecuritySchemes.TOKEN)
 @Tag(ref = RestOpenApiConstants.Tags.DATA)
 @EndpointDisabled(
-    name = "stargate.rest.cql.disabled",
+    name = "stargate.rest.cql-disabled",
```
Comment (tatu-at-datastax) [inline apis/sgv2-restapi/src/main/java/io/stargate/sgv2/restapi/service/resources/CQLResource.java:39]: Looks like Quarkus 3.7 started validating matching of config -- and structure in `RestApiConfig` did not match: it used "stargate.rest.cqlDisabled" (not "stargate.rest.cql.disabled") so system property needs to match wrt capitalization.

File: apis/pom.xml [hunk 2]
```diff
@@ -60,13 +60,13 @@
     <!-- Quarkus -->
     <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
     <quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
-    <quarkus.platform.version>3.6.1</quarkus.platform.version>
+    <quarkus.platform.version>3.11.3</quarkus.platform.version>
```
Comment (kathirsvn) [inline apis/pom.xml:63]: It looks like 3.12 is released now (https://quarkus.io/blog/quarkus-3-12-0-released/). Can we update to the latest?

Comment (tatu-at-datastax) [reply apis/pom.xml:63]: Sure I can try that.

Comment (tatu-at-datastax) [reply apis/pom.xml]: Updated, waiting for CI re-run.

**Review by kathirsvn — APPROVED** [review]

File: apis/pom.xml (hunk 2 above)
Comment (kathirsvn) [reply apis/pom.xml]: CI looks good after upgrade.

**Pull request merged**